### ISSUE TITLE
Swift 2.1 / XCode 7.1.1: Can't subclass Fetcher?

### ISSUE THREAD

**Issue by bes** [Haneke/HanekeSwift#249]
First off: I'm really sorry if I'm misunderstanding something. I would be very happy if I was.

However, I can't seem to subclass Fetcher properly

Haneke version (from Cocoapods): 0.10.0

# What I am trying to do
Most of the http fetch/cache libraries for iOS (particularly swift) seem to completely overlook the need for Authorization headers in the API. All I want to do is add authorization to the request, but I can't seem find any native way of doing it in Haneke. Please correct me if I'm wrong.

My solution to that was that I wanted to implement a Fetcher that would do auth for me, transparently.

However, my attempts were foiled. Stay tuned!

# Variant 1

```swift
import Haneke
class AuthFetcher<T: DataConvertible> : Fetcher<T> {
    init(auth: AnyObject) {
        // Error message: "Super.init isn't called before returning from initalizer". BUT no autocomplete for super.init is available 
    }
}
```

If I manually try to set super.init(key: "whatever") it too fails with an error message, which leads us to...

# Variant 2

```swift
import Haneke

class AuthFetcher<T: DataConvertible> : Fetcher<T> {
    init(auth: AnyObject) {
        super.init(key: "whatever")
        // Error message: "Value of type '_ -> _' has no member 'init'"
    }
}
```
I googled this error message like crazy, but no dice.

# Variant 3
There is no autocomplete override for the fetch method, and when I manually copy the fetch method from NetworkFetcher it looks like this:

```swift
import Haneke
class AuthFetcher<T: DataConvertible> : Fetcher<T> {
    public override func fetch(failure fail : ((NSError?) -> ()), success succeed : (T.Result) -> ()) {
        // Warning message: "Declaring a public instance method for an internal generic class"
    }
}
```

OK Fine, I do what XCode suggests and replace "public" with internal, which leads us to...

# Variant 4

```swift
import Haneke
class AuthFetcher<T: DataConvertible> : Fetcher<T> {
    internal override func fetch(failure fail : ((NSError?) -> ()), success succeed : (T.Result) -> ()) {
        // Error message: "Method does not override any method from its superclass"
    }
}
```
Obviously the XCode suggested fix didn't work

# Conclusion / questions
What am I doing wrong? Is there some fundamental thing that I am completely overlooking?

Thanks in advance!

**Comment by trgoofi** [comment]
@bes `Fetcher` is not public for subclass up to version `0.10.0`.  As you can see from #224 this is the latest commit and it has not been included in `0.10.0`.

**Comment by bes** [comment]
Thanks, makes sense. Hopefully a new version is released to CocoaPods soon.
